File: legacy/src/test/rs/emulate/legacy/archive/ArchiveHashTest.java
package rs.emulate.legacy.archive;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

/**
 * Tests the {@link ArchiveUtils#hash} function.
 *
 * @author Major
 */
public final class ArchiveHashTest {

	/**
	 * The expected hash of the {@link #TEST_NAME} string.
	 */
	private static final int TEST_HASH = 11_943_852;

	/**
	 * The string to hash.
	 */
	private static final String TEST_NAME = "test";

	/**
	 * Executes the test.
	 */
	@Test
	public void test() {
		int hash = ArchiveUtils.hash(TEST_NAME);
		assertEquals(TEST_HASH, hash);
	}

}

File: legacy/src/test/kotlin/rs/emulate/legacy/archive/ArchiveHashTest.kt
package rs.emulate.legacy.archive

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test


/**
 * Tests the [ArchiveUtils.hash] function.
 */
class ArchiveHashTest {

    @Test
    fun test() {
        val hash = ArchiveUtils.hash(TEST_NAME)
        assertEquals(EXPECTED, hash)
    }

    companion object {

        /**
         * The expected hash of the [TEST_NAME] string.
         */
        private const val EXPECTED = 11943852

        /**
         * The string to hash.
         */
        private const val TEST_NAME = "test"
    }

}
